bitpusher module

excalibur0922 opened the pull request:
# Only minor modifications of bitcash allow for the bitpusher module
## bitpusher aims to be a comprehensive solution for:
### 1) Broadcasting custom OP_RETURN pushdata onto the bitcoin cash blockchain
### 2) Easy interface with popular industry standards and protocols such as:
- Tokenisation standards: e.g. SLP (Simple Ledger Protocol - see https://docs.google.com/document/d/1GcDGiVUEa87SIEjrvM9QcCINfoBw-R7EPWzNVR4M8EI);
- Standard registration of Token ID / Issuer ID
- Broadcasting the SHA256 hash of important data / contracts onto the blockchain with your preference of intellectual property archive / industry standard protocol
- Support for a few special cases such as _memo.cash_ and _blockpress.org_ to allow easy access for beginner python developers to gain experience playing around with OP_RETURN-based apps.

## "bitpuller" is also suggested as a complementary counterpart
- This would come in the form of an easy-to-use bitDB query API for python which would synergise well with bitpusher in more than just name!
- Could create an open source, google-like, block-chain search engine here too as part of the overall concept.

### These two modules together would aim to lower barriers to entry for new developers interested in bitcoin cash and make it as easy as possible to: 
- create apps / tokens
- interact with existing apps / tokens
- Do powerful searches / queries of block-chain metadata with an intuitive API.

_Later I would like to package these functions into a Qt GUI for more mainstream use by tech-savvy crypto-enthusiasts. (In a different, stand-alone project)_

teran-mckinney commented on bitcash/wallet.py:
```diff
@@ -238,18 +238,19 @@ def create_transaction(self, outputs, fee=None, leftover=None, combine=True,
         :rtype: ``str``
         """
-
+        print('create:', custom_PUSHDATA)
```
Probably don't want this.

excalibur0922 replied:
indeed not! my bad...

teran-mckinney commented:
Hey!

This is really neat. I definitely need to think over it and review this a bit more. I looked over the code and initially, I think it's mostly okay. You've got that one `print()` in there, and some blank lines with just spaces. I know flake8 isn't used actively yet, but I try to clean that up where I can and plan on making it a standard here.

Also will need to bump the version, probably 0.6.0.

Let me get back to you on this, but if I take too long just ping me. I tend to lose track of things.

Thanks,

Teran

excalibur0922 commented:
Thanks for prompt response. So far, I've:
1) Removed the print() in wallet.py
2) Run flake8 over transactions.py and bitpusher.py --> now compliant (ignoring: E501, w503 codes)
3) Now conforming to PEP8 naming conventions (i.e. changed custom_PUSHDATA variable --> custom_pushdata) and changed PUSHDATA --> pushdata (because it is not a constant)

excalibur0922 commented:
I just broke something... wait...

excalibur0922 commented:
- I think I'm reasonably happy with it now as a basic first installment. Let me know if there are any other changes you'd like. No pressure though if you're just not into it.
- From here, what I might do is continue to build new features on branches: bitpusher_2, bitpusher_3 etc etc. in further bite-sized installments. Don't want to overload you.

teran-mckinney commented:
This is looking great!

Mainly, I want to review the new Bitcoin Cash features, the features you're adding here, also naming considerations and whether or not this makes the most sense in bitcash or not. I think it's really cool, but want to be cautious adding features as that can be a dangerous road in and of itself.

So I'll need a bit more time, but hopefully not more than a few days. Don't want you to wait on me for too long.

And feedback from any other interested parties would be much appreciated.
